File: app/src/main/java/de/exercicse/jrossbach/podcast/network/ApiProvider.java
package de.exercicse.jrossbach.podcast.network;

import org.simpleframework.xml.convert.AnnotationStrategy;
import org.simpleframework.xml.core.Persister;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.simplexml.SimpleXmlConverterFactory;


public class ApiProvider {

    private String baseUrl;
    private Retrofit retrofit;

    public static Retrofit getRetrofit(final String baseUrl){
        return new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(SimpleXmlConverterFactory.createNonStrict(
                        new Persister(new AnnotationStrategy()
                        )))
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();
    }

    public static PodcastApi getChannelApi(final String baseUrl){
        return getRetrofit(baseUrl).create(PodcastApi.class);
    }
}

File: app/src/main/java/de/exercicse/jrossbach/podcast/network/ApiProvider.kt
package de.exercicse.jrossbach.podcast.network

import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class ApiProvider {

    val podcastApi: PodcastApi = getRetrofit().create(PodcastApi::class.java)
    val gson = GsonBuilder().setLenient().create()

    private fun getRetrofit(): Retrofit {
        return Retrofit.Builder()
            .client(OkHttpClient.Builder().addInterceptor(AuthorizationIntercepter()).build())
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .build()
    }

    companion object {
        private val BASE_URL = "https://listen-api.listennotes.com/api/v2/"
    }
}
